sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -428,6 +428,33 @@ class DataFrameSuite extends QueryTest
     }
   }
 
+  test("repartition by MapType") {
+    Seq("int", "long", "float", "double", "decimal(10, 2)", "string", "varchar(6)").foreach { dt =>
```

Comment:
This newly added test case fails at NON-ANSI mode at the test case, `"decimal(10, 2)"`.